src/hooks/useUploadReferral.ts

```diff
@@ -18,10 +21,28 @@ const useUploadReferral = () => {
     const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>) => {
         const file = e.target.files?.item(0);
         if (file && isValidFile(file)) {
-            setFile(file);
+            uploadReferralSheet(usersMock[0].recipientUserId, file).then(() => setFile(file))
         }
     };
 
+    const uploadReferralSheet = (recipientUserId: string, file: File) => {
+        return RecipientUserService.getReferralSheetUploadUrl(recipientUserId)
+            .then((referralSheetUploadUrl) => uploadFileToReferralSheetUploadUrl(referralSheetUploadUrl, file))
+            .then((referralSheetUrl)=> RecipientUserService.updateReferralSheetUrl(recipientUserId, referralSheetUrl))
+    }
+
+    const uploadFileToReferralSheetUploadUrl = (referralSheetUploadUrl: string, file: File) => {
+        return axios.put(referralSheetUploadUrl, file)
```

Comment:
what do you think about extracting the API requests to a client? An example [here](https://levelup.gitconnected.com/enhance-your-http-request-with-axios-and-typescript-f52a6c6c2c8e), but not the only one

Reply:
In the example article is like to make calls to an API with the same baseURL but in my case the baseurl is going to change each time. Then I don't see the advantage there. But I can create a service so the call to an external url is encapsulated there

If I moved this to a service say UploadReferralSheetService it would only have 1 static method like this

```
class UploadReferralSheetService {

      static uploadReferralSheet(referralSheetUploadUrl , file){
          return axios.put(referralSheetUploadUrl, file)
      }
}
```

Is this ok for you?

Reply:
I would abstract this and the `getReferralSheetUrlFromReferralSheetUploadUrl` into a `ReferralSheetService`